cocos/renderer/CCGLProgramState.cpp

```diff
@@ -165,8 +175,24 @@ void UniformValue::setTexture(GLuint textureId, GLuint textureUnit)
     //CCASSERT(_uniform->type == GL_SAMPLER_2D, "Wrong type. expecting GL_SAMPLER_2D");
     _value.tex.textureId = textureId;
     _value.tex.textureUnit = textureUnit;
+    _value.tex.texture = nullptr;
+    _type = Type::VALUE;
+}
+
+void UniformValue::setTexture(Texture2D* texture, GLuint textureUnit)
+{
+    CCASSERT(texture != nullptr, "texture is nullptr");
+
```

Comment:
could it be possible that someone calls:

```c++
setTexture(someTexture, 0);
setTexture(someTexture, 1);
```
because, if that is the case, the 2nd call won't do anyting